sql/server_component/log_sink_trad.cc

```diff
@@ -349,8 +350,23 @@ int log_sink_trad(void *instance MY_ATTRIBUTE((unused)), log_line *ll) {
         log_sink_trad_last = microtime;
       }
 
+      bool log_to_buffered_log = false;
+      bool log_only_to_buffered_log = false;
```

Comment:
log_only_to_buffered_log

Reply:
No, these both have real usecases.

log_to_buffered_log tells us if we should write something to the buffered log.
log_only_to_buffered_log tells us that we shouldn't write that message to the normal error log, only to the buffered log.

We definitely need log_only_to_buffered_log. 

Log_to_buffered_log is a different question. For performance testing, I've written all log messages to the buffered error log too, but we might have use cases where we only want to write some of the log messages there. The mariadb implementation is like that (based on the documentation), they only write wsrep messages to the blackbox, that's why both variables are here.

Reply:
I mean, it will be compiler optimized anyway. I see the variable used
bool x  = false
if (!false || ..)

Reply:
Because in the current code, there's nothing using this logging other than that DBUG_EXECUTE_IF code (and that only checks that we write to the buffered log correctly).

PXC will set log_only_to_buffered_log true if we are writing a wsrep/galera log message above the warning level.

Reply:
So if I understand correctly, there are two functionalities we want to achieve?
1. Bufffered logging (log to buffer and write to disk when buffer is full)
2. Buffer-only Logging (write only to buffer, when buffer is full, we discard/overwrite. Buffer is written to disk only at crash)?

Reply:
And regarding 1. the buffered logging, can't we use same error log file ? ie. write buffered messages to default mysqld error log.

Reply:
If not for log rotation, I would love to see a simpler logic. Just use buffered functionality for all messages and write buffer to mysqld error log. 

Maybe we should keep the log rotation problem separate and fix that separately. WIth that my proposal would be

1. No separate error log file (no option for this)
2. If buferred_error_log_buffer_size is not 0, use buffering and write the buffer to mysqld error log file.
3. If buferred_error_log_buffer_size is 0, normal logging (as if this feature is disabled)

Reply:
@dutow what do u think? u want to keep buffered log file as separate?

Reply:
Yes, I'd like to keep it separe for several reasons:

1. buffered logging is delayed. Users might want to reveive normal log messages without significant delays.
2. buffered logging is intended for debug logs (high throughput, huge logfiles). This is only required if something bad happens, and we want to investigate how that happened, while the normal logs could be useful for normal server operation. Also, because of the larger size, keeping it separate means users can use different rotation/size limits (e.g. keep normal logs for a few months, but keep buffered logs only for the last day, or something like that)

Reply:
ok @dutow I think the answers are convincing enough to keep it as is